app/src/main/java/com/android/wildex/ui/social/FriendScreenViewModel.kt

```diff
@@ -0,0 +1,351 @@
+        // not enough left
+        if (state.isCurrentUser && suggestions.value.size < 5) {
+          suggestions.value = userRecommender.getRecommendedUsers()
+        }
+      } catch (e: Exception) {
+        _uiState.value = state
+        setErrorMsg("Failed to send request to user $userId : ${e.message}")
+        // if the request was done in the suggestions, we need to retrieve the suggestion
+        if (state.isCurrentUser) suggestions.value = state.suggestions
+      }
+    }
+  }
+
+  /**
+   * Unfollows the given user. Unfollowing is always possible whether we are viewing the current
+   * user's friend screen or that of another user and it results in the same effect in both views.
+   * Unfollowing a user means updating the friends states to reflect the change in the screen, i.e.
+   * notify that the given user was unfollowed and allowing the current user to send a new friend
+   * request if the action was accidental.
+   *
+   * @param userId the user who the current user wishes to unfollow
+   */
+  fun unfollowUser(userId: Id) {
+    viewModelScope.launch {
+      val state = _uiState.value
+
+      val friends = state.friends
+      val newFriends =
+          friends.map {
+            if (it.friend.userId == userId) it.copy(isFriend = false, isPending = false) else it
+          }
+
+      _uiState.value = state.copy(friends = newFriends)
+
+      try {
+        userFriendsRepository.deleteFriendToUserFriendsOfUser(userId, currentUserId)
+        userFriendsRepository.deleteFriendToUserFriendsOfUser(currentUserId, userId)
+      } catch (e: Exception) {
+        _uiState.value = state
+        setErrorMsg("Failed to unfollow user $userId : ${e.message}")
+      }
+    }
+  }
+
+  /**
+   * Accepts a request received by the current user. This operation is only possible when the screen
+   * is the current user's friend screen. Thus, accepting a received friend request always results
+   * in the friend request disappearing from the received requests section and a new friend
+   * appearing in the current user's friend list
+   *
+   * @param userId the user whose friend request is accepted
+   */
+  fun acceptReceivedRequest(userId: Id) {
+    viewModelScope.launch {
+      val state = _uiState.value
+
+      val receivedRequests = state.receivedRequests
+      val newReceivedRequests = receivedRequests.filter { it.senderId != userId }
+
+      val friends = state.friends
+      val newFriends =
+          friends +
+              listOf(
+                  FriendState(
+                      friend = userRepository.getSimpleUser(userId),
+                      isFriend = true,
+                      isPending = false,
+                      isCurrentUser = false))
+
+      _uiState.value = state.copy(friends = newFriends, receivedRequests = newReceivedRequests)
+
+      try {
+        friendRequestRepository.acceptRelationship(Relationship(userId, currentUserId))
```

Comment:
Use the future *acceptFriendRequest(friendRequest: FriendRequest)*
```suggestion
        friendRequestRepository.acceptFriendRequest(Relationship(userId, currentUserId))
```